feat(images): Adding iOS Image Cache Limits (#3258)
facebook/react-native-website: docs/images.md

## Off-thread Decoding

Image decoding can take more than a frame-worth of time. This is one of the major sources of frame drops on the web because decoding is done in the main thread. In React Native, image decoding is done in a different thread. In practice, you already need to handle the case when the image is not downloaded yet, so displaying the placeholder for a few more frames while it is decoding does not require any code change.

## Configuring iOS Image Cache Limits

> 

Reviewer: Ideally we can include 1-2 sentences of context here. Roughly:

> On iOS, we expose an API to override React Native's default image cache limits. This should be called from within your native `AppDelegate` code (e.g. within (e.g. within `didFinishLaunchingWithOptions`).

```objectivec
RCTSetImageCacheLimits(4*1024*1024, 200*1024*1024);
```

**Parameters:**

| Name           | Type   | Required | Description             |
| -------------- | ------ | -------- | ----------------------- |
| imageSizeLimit | number | Yes      | Image cache size limit. |
| totalCostLimit | number | Yes      | Total cache cost limit. |

> Based on the code above the single image size limit would be set to 4 MB and the total cost limit would be set to 200 MB.

Reviewer: ```suggestion
In the above code example the image size limit is set to 4 MB and the total cost limit is set to 200 MB.
```

## Configuring iOS Image Cache Limits

> On iOS, we expose an API to override React Native's default image cache limits. This should be called from within your native AppDelegate code (e.g. within (e.g. within didFinishLaunchingWithOptions).

Reviewer 2: ```suggestion
On iOS, we expose an API to override React Native's default image cache limits. This should be called from within your native AppDelegate code (e.g. within `didFinishLaunchingWithOptions`).
```
Small nit.
